trin-core/src/portalnet/overlay_service.rs

```diff
@@ -1812,8 +2042,71 @@ mod tests {
         let expected_index = bucket.get_index(&key).unwrap();
         assert_eq!(target_bucket_idx, expected_index as u8);
     }
+
+    #[rstest]
+    #[case(3)]
+    #[case(7)]
+    #[case(8)]
+    #[case(17)]
+    #[case(25)]
+    fn test_limit_nodes_response_size(#[case] original_nodes_size: u8) {
+        let mut enrs: Vec<SszEnr> = Vec::new();
+        for _ in 0..original_nodes_size {
+            // Generates an ENR of size 63 bytes.
+            let (_, enr) = generate_random_remote_enr();
+            enrs.push(SszEnr::new(enr));
+        }
+
+        let enrs_limited = limit_nodes_response_size(enrs);
+
+        let correct_limited_size = match original_nodes_size {
```

Comment:
I think you can just add a second argument to the parameterization... [like here](https://github.com/ethereum/trin/blob/37c1931a2f2eb9e4832d980574070e8ff1a5311e/trin-core/src/utils/bootnodes.rs#L54)